### PR TITLE
Mention Streams and Lambda's

### PR DESCRIPTION
**Pull request by lesv**
Also mention java 8 / Java 11

### CODE REVIEW

File: SAMPLE_FORMAT.md [hunk 1]
```diff
@@ -199,6 +202,16 @@ Snippets should have tests that should verify the snippet works and compiles cor
 The following are some general Java best practices that should be followed in samples to remain
 idiomatic. 
 
+### Modern Java
```
Comment (kurtisvg) [inline SAMPLE_FORMAT.md]: In order to make these easier to reference, can we separate them into different sections? It sounds like there are three sections:

- Labdas
- Streams
- For-each loops over `ArrayLists.forEach()`

File: SAMPLE_FORMAT.md [hunk 2]
```diff
@@ -199,6 +202,16 @@ Snippets should have tests that should verify the snippet works and compiles cor
...
+Prefer using modern ideoms / language features over older styles.
```
Comment (averikitsch) [inline SAMPLE_FORMAT.md]:
```suggestion
Prefer using modern idioms / language features over older styles.
```

File: SAMPLE_FORMAT.md [hunk 3]
```diff
@@ -199,6 +202,16 @@ Snippets should have tests that should verify the snippet works and compiles cor
...
+
+* Lambda's should be about 1-3 lines at most, otherwise it should be in a named method.
+  * Prefer to annonymous classes
+* Streams can be extremely compact, efficient, and easy to use - consider using them.
+  * Avoid side effects
```
Comment (averikitsch) [inline SAMPLE_FORMAT.md]: nit: can you expand on this?

Comment (lesv) [reply SAMPLE_FORMAT.md]: I was trying to make it as short as possible, but from the internal java guidelines (see meeting doc).
That said, are you asking me to explain "side effects"?

**Comment by lesv** [comment]
PTAL

File: SAMPLE_FORMAT.md [hunk 4]
```diff
@@ -194,12 +197,27 @@ Snippets should have tests that should verify the snippet works and compiles cor
  verify that it works correctly. See the tests in the canonical for an example of how to do this
  correctly. 
 
+## Modern Java
+Prefer using modern idioms / language features over older styles.
+
+### Lambda's
+Should be about 1-3 lines at most, otherwise it should be in a named method.
+* Prefer lambda's to annonymous classes
```
Comment (averikitsch) [inline SAMPLE_FORMAT.md]: nit:
```suggestion
* Prefer lambdas to annonymous classes
```

**Comment by lesv** [comment]
@dpebot merge when green

**Comment by lesv** [comment]
Since this is just a readme - I'm going to merge this.

**Pull request merged**